### DISCUSSION TITLE
Intellisense and auto-completion take ages and saving fiels with eslint and prettier is very slow

### DISCUSSION THREAD

**Discussion by Saamuuell** [vuejs/language-tools#4938] [Q&A]
Am I the only one this happens to? Recently since the Eslint update saving a file can take up to 10 seconds even with only Vue, Eslint and prettier as extension. Also, I don't know when it started to appear, but auto-completion on vscode is getting slower and slower, with sometimes 'no-suggestions' after loading for 10 seconds. I tried everything to speed up the process. I'm at performance levels which are acceptable but far from when the Volar extension was not deprecated.

Not only that, but I also tried to increase the ram that vscode can use, but since some updates, it is impossible to do so.

My vscode config to speed this up a little bit: 
```json
{
  "editor.defaultFormatter": "esbenp.prettier-vscode",
  "eslint.codeActionsOnSave.rules": null,
  "eslint.enable": true,
  "editor.codeActionsOnSave": [
    "source.organizeImports",
    "source.fixAll.eslint"
  ],
  "eslint.codeActionsOnSave.mode": "problems",
  "eslint.validate": ["typescript"],
  "typescript.format.enable": false,
  "javascript.format.enable": false,
   "vue.server.maxOldSpaceSize": 4096,
   "typescript.tsserver.maxTsServerMemory": 4096,
    "files.watcherExclude": {
      "**/node_modules/**": true,
      "**/dist": true,
      "**/.git": true,
      "**/public": true,
   },
   "editor.wordBasedSuggestions": "currentDocument",
} 
```

**Comment by AemonCao** [comment]
I have the same problem, and sometimes it saves fine, and sometimes it doesn't save at all, even for the same project.

![image](https://github.com/user-attachments/assets/b4765994-1bee-4cbc-a03b-7ec74a1981b3)

I tried disabling all plugins, but as long as Vue-Official and ESLint are present at the same time, and the following settings are enabled:

```json
"editor.codeActionsOnSave": {
    "source.fixAll.eslint": "explicit",
    "source.organizeImports": "never"
},
```

The problem occurs.

**Comment by s3xybun** [comment]
Me too. I'm using eslint + vue-official. Sometimes it is very slow 😥.

**Reply by AemonCao** [reply]
昨天研究了一下，感觉是 ESLint 有时候没有启动，可以在Ctrl+Shift+P运行一下 `ESLint: Restart ESLint Server`，然后再看一下 `ESLint: Show Output Channel` 日志，看看报错提示。

之后设置一下默认的格式化程序：

```JSON
"[javascript]": {
    "editor.defaultFormatter": "dbaeumer.vscode-eslint"
},
"[typescript]": {
    "editor.defaultFormatter": "dbaeumer.vscode-eslint"
},
"[vue]": {
    "editor.defaultFormatter": "dbaeumer.vscode-eslint"
},
```

然后关闭 Vue 插件的 CodeActions：

```JSON
"vue.codeActions.enabled": false,
```

这样设置后，在我的项目上是可以正常 Lint 了，虽然还是会出现问题，但是只要运行一下 `ESLint: Restart ESLint Server`，并保证 ESLint 服务能正常启动，就可以顺利保存。

---

Yesterday, I looked into the issue and found that sometimes ESLint doesn’t start automatically. You can try running `ESLint: Restart ESLint Server` from the Command Palette (Ctrl+Shift+P), then check the `ESLint: Show Output Channel` logs to see any error messages.

Next, set the default formatter as follows:

```JSON
"[javascript]": {
    "editor.defaultFormatter": "dbaeumer.vscode-eslint"
},
"[typescript]": {
    "editor.defaultFormatter": "dbaeumer.vscode-eslint"
},
"[vue]": {
    "editor.defaultFormatter": "dbaeumer.vscode-eslint"
},
```

Then, disable the Vue plugin’s CodeActions:

```JSON
"vue.codeActions.enabled": false,
```

With these settings, ESLint runs normally in my project. Although some issues still come up, running `ESLint: Restart ESLint Server` and ensuring the ESLint server is running allows it to save without problems.

**Reply by s3xybun** [reply]
Downgrade from v2.1.8 to v2.1.6 works for me.
—-
Nope. Still slow. I doubt it is a issue of vscode.
